addons/mail/models/mail_channel.py

```diff
@@ -726,13 +726,18 @@ def channel_pin(self, uuid, pinned=False):
         channel = self.search([('uuid', '=', uuid)])
         channel._execute_channel_pin(pinned)
 
+    def channel_pin_for_all_members(self):
+        """
+        Pin channels for every members of each.
+        """
+        self.mapped('channel_last_seen_partner_ids').write({'is_pinned': True})
```

Comment:
No need to call `mapped` to follow a relational field.

Note: this might actually be much slower than using SQL precisely due to having to read all channels partners of all channels in self just to write on them. Not sure if we need to care about perf at this point (not even sure yet if we'll keep the instruction), but something to keep in mind.